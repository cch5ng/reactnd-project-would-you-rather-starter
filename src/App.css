* {
  margin: 0;
  padding: 0;
  font-family: 'Poppins', sans-serif;
  font-weight: 300;
}

.App {
  /*text-align: center;*/
  margin: 0 5% 20px 5%;
}

h1,
h2,
h3,
h4 {
  font-weight: 400;
}

/*@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
*/
/* Nav */

.header {
  background-color: #1f2035;
  box-sizing: border-box;
  /*color: white;*/
}

/*.title:after {
  clear: both;
}*/

.header-main,
.header-contact {
  color: #f5ffa6; /*#d8f2e1;*/
  width: 100%;
  text-align: center;
  text-decoration: none;
}

.contact {
  /*color: #6666e0;*/
  /*font-size: 1.35em;*/
  padding: 12px 0;
  text-align: center;
}

.contact a:link,
.contact a:visited {
  color: #fff;
  text-decoration: none;
  display: block;
}

/* Form */
.form-container {
  background-color: #cceae7;
  padding: 15px;
}

.button {
  height: 25px;
  font-size: 14px;
  margin-right: 3px;
  padding: 0 5px;
  border-radius: 5px;
  border-style: none;
  background-color: #fff;
}

.button:hover {
  background-color: #f4fafa;
}

input {
  width: calc(100% - 34px);
  height: 25px;
  padding-left: 8px;
  font-size: 14px;
  border-radius: 5px;
  border: 1px solid #999;
}

textarea {
  width: calc(100% - 34px);
  height: 60px;
  padding-left: 8px;  
  font-size: 14px;
  border-radius: 5px;
  border: 1px solid #999;
}

select,
option {
  height: 25px;
  font-size: 14px;  
}

input:focus,
select:focus,
option:focus,
textarea:focus {
  outline: none !important;
  border:1px solid #c0e585; /*#d6ff94;*/
  box-shadow: 0 0 10px #c0e585; /*#d6ff94; #d6ff94;*/
}

@media (min-width: 768px) {

  .header {
    background-color: #1f2035;
    height: 70px;
  }

  .header-main,
  .header-contact {
    float: left;
    display: inline-block;
  }

  .header-main,
  .header-contact {
    width: calc(50% - 30px / 2);
  }

  .header-main {
    margin-top: 15px;
    margin-left: 15px;
    text-align: left;
  }

  .header-contact {
    text-align: right;
    vertical-align: top;
    margin-top: 8px;
    margin-right: 15px;
  }

  .contact {
    padding-bottom: 12px;
    text-align: right;
  }

  .contact a:link,
  .contact a:visited {
    padding-left: 12px;
    display: inline;
  }

  .col-50p {
    width: 50%;
  }

}
