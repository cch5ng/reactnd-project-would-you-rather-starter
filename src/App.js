import React, { Component } from 'react';
import {
  BrowserRouter as Router, Route, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './App.css';
import { receiveLogin, receiveLogout } from './login/loginActions';
import { fetchUsers } from './users/usersActions';
import { fetchQuestions } from './questions/questionsActions';
import Questions from './questions/Questions';
import QuestionDetail from './questions/QuestionDetail';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loginUser: '',
      isLoggedIn: false
    }

    this.onChangeHandler = this.onChangeHandler.bind(this);
    this.logout = this.logout.bind(this);
    this.makeArrayFromDictionary = this.makeArrayFromDictionary.bind(this);
  }

  componentDidMount() {
    this.props.dispatch(fetchUsers());
    this.props.dispatch(fetchQuestions());
  }

  onChangeHandler(ev) {
    let value = ev.target.value;
    let stateChange = {};
    stateChange[ev.target.name] = value;
    this.setState(stateChange, function() {
      if (this.state.loginUser.length) {
        // change state to logged in
        console.log('user is logged in');
        this.props.dispatch(receiveLogin(this.state.loginUser));
      }
    });
  }

  logout(ev) {
    console.log('clicked logout');
    this.setState({loginUser: ''});
    this.props.dispatch(receiveLogout());
  }

  // getUserDictionary() {
  //   return {'sarahedo': 'Sarah Edo',
  //           'tylermcginnis': 'Tyler McGinnis',
  //           'johndoe': 'John Doe'
  //          }
  // }

  makeArrayFromDictionary(dict) {
    return Object.keys(dict).map(uid => dict[uid]);
  }

/*


          <Route exact path="/" render={() => (
              <Questions />
            )}
          />


          <Route exact path="/add" render={() => (
              <Questions />
            )}
          />

          <Route exact path="/leaderboard" render={() => (
              <Questions />
            )}
          />

*/

  render() {
    let isLoggedIn;
    let userDictionary;
    let curUserId;
    let curUser;
    let userAr;
    let questionDictionary;
    let userAnswers;
    let userQuestions;

    if (this.props.users && this.props.users.users) {
      userDictionary = this.props.users.users;
      userAr = this.makeArrayFromDictionary(userDictionary);
    }

    // if user logged in, get user name and user questions
    if (this.props.login) {
      isLoggedIn = this.props.login['isLoggedIn'];
      if (isLoggedIn) {
        let loggedInUid = this.state['loginUser'];
        if (userDictionary) {
          curUser = userDictionary[loggedInUid]['name'];
        }
        if (this.props.questions && this.props.questions.questions) {
          questionDictionary = this.props.questions.questions;
          console.log('questionDictionary: ' + JSON.stringify(questionDictionary));
          userAnswers = userDictionary[loggedInUid]['answers'];
          userQuestions = userDictionary[loggedInUid]['questions'];

          console.log('userAnswers: ' + JSON.stringify(userAnswers));

        }
      }
    }

    return (
      <Router>
        <div className="App">
          <header className="header title">
            <div className="header-main">
              <h2><Link to="/" className="header-main">Would You Rather</Link></h2>
            </div>
            <div className="header-contact">
              <p className="contact">
                <Link to="/">Questions</Link>
                <Link to="/leaderboard">Leader Board</Link>
              </p>
              {isLoggedIn === false && (
                <select value={this.state.loginUser} onChange={this.onChangeHandler} name="loginUser">
                  <option value="">Login</option>
                  {userAr.map(user => {
                    return (<option value={user["id"]} key={user["id"]}>{user["name"]}</option>)
                  })}
                </select>
              )}
              {isLoggedIn === true && (
                <Link to="/" onClick={this.logout} >Logout {curUser}</Link>
              )}
            </div>
          </header>

          <main>
            <Route exact path="/" render={() => (
                <Questions isLoggedIn={isLoggedIn} 
                  userQuestions={userQuestions} 
                  userAnswers={userAnswers}
                  questions={questionDictionary}
                />
              )}
            />
            <Route exact path="/questions/:question_id" render={({match}) => (
                <QuestionDetail match={match} userDictionary={userDictionary} />
              )}
            />
            <Route exact path="/leaderboard" render={() => (
                <div>test data</div>
              )}
            />


          </main>

          <footer>Link to source
          </footer>
        </div>
      </Router>
    );
  }
}

function mapStateToProps({ login, users, questions }) {
  return {
    login,
    users,
    questions
  }
}

export default connect(mapStateToProps)(App);
