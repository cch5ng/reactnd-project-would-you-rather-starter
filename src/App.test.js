import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { App } from './App';

const mockReceiveLogin = jest.fn();
const mockReceiveLogout = jest.fn();
const mockFetchUsers = jest.fn();
const mockFetchQuestions = jest.fn();

it('renders without crashing', () => {
  let wrapper;
  const div = document.createElement('div');

  ReactDOM.render(<App receiveLogin={mockReceiveLogin} receiveLogout={mockReceiveLogout}
    fetchUsers={mockFetchUsers} fetchQuestions={mockFetchQuestions} 
    />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('App renders', () => {
  // snapshot test
  const tree = renderer
    .create(<App receiveLogin={mockReceiveLogin} receiveLogout={mockReceiveLogout}
        fetchUsers={mockFetchUsers} fetchQuestions={mockFetchQuestions}
      />,)
    .toJSON();

    expect(tree).toMatchSnapshot();
});
