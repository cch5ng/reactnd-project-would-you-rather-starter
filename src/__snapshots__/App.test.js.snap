// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="App"
>
  <header
    className="header title"
  >
    <div
      className="header-main"
    >
      <h2>
        <a
          className="header-main"
          href="/"
          onClick={[Function]}
        >
          Would You Rather
        </a>
      </h2>
    </div>
    <div
      className="header-contact"
    >
      <p
        className="contact"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Questions
        </a>
        <a
          href="/add"
          onClick={[Function]}
        >
          New Question
        </a>
        <a
          href="/leaderboard"
          onClick={[Function]}
        >
          Leader Board
        </a>
      </p>
    </div>
  </header>
  <main>
    <div
      className=""
    >
      <h1>
        Questions
      </h1>
      <div>
        Sorry, you need to log in to view questions.
      </div>
    </div>
  </main>
  <footer>
    Link to source
  </footer>
</div>
`;
